li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/libangbang/LiHadoopShellCommand.groovy

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.gradle.libangbang;
+
+/**
+ * All the command for jobtypes which support bangbang should extend this class
+ */
+public abstract class LiHadoopShellCommand {
```

Comment:
I'm wondering if we should name this something else - perhaps LiBangBangCommand? What I'm concerned about is that LiHadoopShellCommand doesn't extend HadoopShellCommand. If it did, then yes I would think the name should be LiHadoopShellCommand.

Similarly for the issue of naming LiHadoopShellCommandFactory -> LiBangBangCommandFactory.

What do you think?